### PR TITLE
Display fills in a stable order

### LINKED ISSUE #1492: Show Fills in a stable order
The Fills should be displayed in a stable order. Currently, we have the prioritized default groups and then the other groups are handled in a first come, first served manner. To allow the extension authors to precisely place their own fills, we should have a stable ordering rules here.

Currently existing groups:
- **general**
- **save**
- **editor**
- **export**
- color
- align
- distribute
- deploy

**bold** - prioritized

### PR DESCRIPTION
**Pull request by barmac**
Fills are now ordered alphabetically by group name and take
a new property `priority` with 0 as default which is used
to determine the order within a group. The higher the priority,
the earlier the Fill is displayed.

Closes #1492

### CODE REVIEW

File: client/src/app/slot-fill/__tests__/SlotFillSpec.js [hunk 1]
```diff
@@ -210,6 +211,81 @@ describe('slot-fill', function() {
...
+
+      });
+
+
+      it('should display fills ordered by priority inside same group', function() {
+
+        // when
+        var slotFillRoot = ReactDOM.render(
+          <SlotFillRoot>
+            <Fill name="foo" group="1_a" priority={ -1 }>
```
Comment (nikku) [inline client/src/app/slot-fill/__tests__/SlotFillSpec.js]: As we are reworking the way fills work, does it make sense to rename `Fill#name` to `Fill#slot`? Looking at the code nowadays I find the `name` property very unintuitive.

Comment (barmac) [reply client/src/app/slot-fill/__tests__/SlotFillSpec.js]: Sounds reasonable. `name` seems to suggest that it's a fill's name when in fact it is a slot's name. I'll have another commit for that.

Comment (barmac) [reply client/src/app/slot-fill/__tests__/SlotFillSpec.js]: Done.

Comment (nikku) [reply client/src/app/slot-fill/__tests__/SlotFillSpec.js]: Much better now, thanks for the additional update.

**Pull request merged**